components/chatUI/ChatInput.tsx

```diff
@@ -0,0 +1,96 @@
+import { SendIcon } from "lucide-react";
+import { FormEvent, useRef, useState } from "react";
+import { MdAttachFile } from "react-icons/md";
+import { VscStopCircle } from "react-icons/vsc";
+
+interface ChatInputProps {
+  input: string;
+  handleInputChange: (
+    e:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
+  ) => void;
+  handleSubmit: (
+    e: React.FormEvent<HTMLFormElement>,
+    options?: {
+      experimental_attachments?: FileList;
+    }
+  ) => Promise<void>;
+  isLoading: boolean;
+  stop: () => void;
+}
+
+export function ChatInput({
+  input,
+  handleInputChange,
+  handleSubmit,
+  isLoading,
+  stop,
+}: ChatInputProps) {
+  const [files, setFiles] = useState<FileList | undefined>(undefined);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const onSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    handleSubmit(event, { experimental_attachments: files });
+
+    setFiles(undefined);
+
+    if (fileInputRef.current) {
+      fileInputRef.current.value = "";
+    }
+  };
+  return (
+    <div className="bg-white p-4">
+      <form onSubmit={onSubmit} className="max-w-4xl mx-auto">
+        <div className="relative flex items-center">
+          <input
+            className="w-full p-3 pr-20 border border-gray-200 rounded-lg 
+              text-black focus:outline-none focus:border-blue-500 focus:ring-1 focus:ring-blue-500"
+            value={input}
+            placeholder="Type your message..."
+            onChange={handleInputChange}
```

Comment:
Add onKeyDown listener and trigger submit on enter. There is noticable delay between enter press and message submission.